sklearn_hierarchical_classification/tests/fixtures.py

```diff
@@ -96,6 +141,50 @@ def make_classifier_and_data(
     return clf, (X, y)
 
 
+def make_classifier_and_data_own_preprocessing():
+    """Create data set and classifier for testing a multi-label
+    classification scenario with a preprocessing pipeline.
+
+    """
+    newsgroups_train = fetch_20newsgroups(subset="train")
+    X, Y = newsgroups_train.data, newsgroups_train.target
+    class_hierarchy = make_newsgroups_hierarchy()
+    names = newsgroups_train.target_names
+
+    vectorizer = TfidfVectorizer(
+        strip_accents=None,
+        lowercase=True,
+        analyzer="word",
+        ngram_range=(1, 3),
+        max_df=1.0,
+        min_df=0.0,
+        binary=False,
+        use_idf=True,
+        smooth_idf=True,
+        sublinear_tf=True,
+        max_features=70000
+    )
+    bclf = OneVsRestClassifier(LinearSVC())
```

Comment:
ping @fbenites  - I am very close to finalizing this PR and merging it including all your awesome work.

One pending item for me to understand - it seems like your work implicitly puts the classifier into a "LCN" (Local Classifier per Node) mode. e.g. instead of having a multi-class classifier per parent node ("LCPN"), we have a single binary classifier per node which tells you if this node should be considered as a predicated label or not.

Is this right? if so I will make some further updates / unit-tests to make sure the mlb/preprocessing flags work in both LCPN and LCN modes, as well as make the support for LCN more explicit

Reply:
It should be independent, the preprocessing can be made either way. We could create also a test with "LCPN".